### DISCUSSION TITLE
Can a mise plugin depend on other mise plugins?

### DISCUSSION THREAD

**Discussion by hisaac** [mise-plugins/.github#15] [Q&A]
I'd like to try my hand at developing a mise plugin for managing Xcode versions. Unfortunately, Xcode releases are done in a pretty non-standard way. Apple doesn't publish version information on GitHub or in git tags (it's tracked manually by the community at https://xcodereleases.com), and installing Xcode versions is a non-trivial process, requiring authentication with an Apple ID.

Relying on a couple dependencies would make an Xcode version plugin way easier. Specifically, using `jq` to parse the JSON api provided by xcodereleases.com and `xcodes` to handle authentication and installation of Xcode versions.

Is there any precedent for a mise plugin depending on other mise plugins?

**Comment by hisaac** [comment]
cc @pepicrft

**Comment by jdx** [comment]
yes but they are currently hardcoded

https://github.com/jdx/mise/blob/f71e3ef047d91b69eb4eb009536ea820c23a2e4b/src/backend/asdf.rs#L291-L298

**Comment by pepicrft** [comment]
> Relying on a couple dependencies would make an Xcode version plugin way easier. Specifically, using jq to parse the JSON api provided by xcodereleases.com and xcodes to handle authentication and installation of Xcode versions.

Using `jq` is better than treating the output as a string and regex'ing against it, but assuming the schema should not change, I wonder if you can instead do:

```
curl -s https://xcodereleases.com/data.json | grep -o '"number":"[0-9.]*"' | sed 's/"number":"\([0-9.]*\)"/\1/'
```

Similarly, how involved is the installation? It's been a long time since I last had to automate the installation in a macOS environment, but I don't remember it being many lines of code. 

My point is that it might be possible to install it without dependencies.

**Reply by hisaac** [reply]
Good point about using `grep` and friends in lieu of `jq`. Not as easy or as powerful, but it could likely work for this case.

As for downloading Xcode, the problem is authentication — you have to be signed into a Apple ID that has access to the Apple Developer Portal. Maybe that's easier to do in a shell script than I'd expect, but I always assume anything auth-related is going to be a pain, and rely on existing tools that have already figured it out.

**Reply by hisaac** [reply]
I've also wondered if one could include a compiled binary instead of scripts. A little Swift or Go CLI could accomplish this nicely.

**Reply by jdx** [reply]
We have initial support for vfox plugins in mise which might have more of the flexibility you're looking for
